lib/internal/test_runner/runner.js

```diff
@@ -123,6 +123,9 @@ function getRunArgs(path, { inspectPort, testNamePatterns, only }) {
   if (only === true) {
     ArrayPrototypePush(argv, '--test-only');
   }
+  if (coverage === true) {
```

Comment:
instead of adding the flag here, we should modify `filterExecArgv` to not filter `--experimental-test-coverage` in case `opts.watch === true`